src/main/java/hudson/plugins/jira/selector/perforce/PerforceJobIssueSelector.java

```diff
@@ -0,0 +1,78 @@
+package hudson.plugins.jira.selector.perforce;
+
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import com.tek42.perforce.model.Changelist;
+import com.tek42.perforce.model.Changelist.JobEntry;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.jira.JiraSite;
+import hudson.plugins.jira.Messages;
+import hudson.plugins.jira.RunScmChangeExtractor;
+import hudson.plugins.jira.selector.AbstractIssueSelector;
+import hudson.plugins.perforce.PerforceChangeLogEntry;
+import hudson.scm.ChangeLogSet;
+import hudson.scm.ChangeLogSet.Entry;
+
+@Extension(optional = true)
```

Comment:
Per documentation:
```
 * Marks a field, a method, or a class for automatic discovery, so that Hudson can locate
 * implementations of {@link ExtensionPoint}s automatically.
 *
 * <p>
 * (In contrast, in earlier Hudson, the registration was manual.)
 *
 * <p>
 * In a simplest case, put this on your class, and Hudson will create an instance of it
 * and register it to the appropriate {@link ExtensionList}.
 *
 * <p>
 * If you'd like Hudson to call
 * a factory method instead of a constructor, put this annotation on your static factory
 * method. Hudson will invoke it and if the method returns a non-null instance, it'll be
 * registered. The return type of the method is used to determine which {@link ExtensionList}
 * will get the instance.
 *
 * Finally, you can put this annotation on a static field if the field contains a reference
 * to an instance that you'd like to register.
 *
```

See DefaultIssueSelector where @Extension is only at DescriptorImpl level.